km/km_vcpu_run.c

```diff
@@ -435,8 +436,15 @@ static int hypercall(km_vcpu_t* vcpu, int* hc_ret)
    if (km_collect_hc_stats != 0) {
       clock_gettime(CLOCK_MONOTONIC, &start);
    }
-   km_hc_ret_t ret = km_hcalls_table[hc](vcpu, hc, ga_kma);
-   *hc_ret = ga_kma->hc_ret;
+   km_hc_ret_t ret = HC_CONTINUE;
+   if (km_hcalls_table[hc] != NULL) {
+      ret = km_hcalls_table[hc](vcpu, hc, ga_kma);
+      *hc_ret = ga_kma->hc_ret;
+   } else {
+      km_infox(KM_TRACE_HC, "Unimplemented hypercall %d (%s)", hc, km_hc_name_get(hc));
+      ga_kma->hc_ret = (uint64_t) -ENOTSUP;
```

Comment:
Maybe we could return ENOSYS so they know it is the system call that is not supported, not some unsupported argument to the system call.

Reply:
OK. I can do ENOSYS instead of ENOTSUP.

Reply:
I'm changing it back since it seems some important user land frameworks handle `ENOTSUP` better than `ENOSYS`.